### PR TITLE
update test docs

### PR DESCRIPTION
**Pull request by creviera**
Remove reference to old TOTP code and update README around functional testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -276,55 +276,94 @@ An AppArmor profile is available for mandatory access control. When installing s
...
-3. Set up the dev server with data required for functional tests to pass and run in any order:
-    - Create two new sources, each with one submission that contains both a file and a message. The message should be set to `this is the message`. The file should be called `hello.txt` and contain a single line of text: `hello`.
-3. Delete all the old cassettes by running `rm -r tests/functional/cassettes` or just delete the cassettes you wish to regenerate.
-4. Run `make test-functional` which will regenerate cassettes for any that are missing in the `tests/functional/cassettes` folder.
+To set up a local dev server and generate cassettes, follow these steps:
+
+1. Bypass TOTP verification so that we can use the TOTP value of `123456` hard-coded in `tests/conftest.py`. You can do this by applying the following diff to the server code:
+
+   ```diff
+   diff --git a/securedrop/models.py b/securedrop/models.py
```
Comment (emkll) [inline README.md:334]: this diff doesn't apply cleanly for me, with the error:
```
error: corrupt patch at line 13
```

The following diff did work for me, but looking at the previous revision, the following did appear to work for me.:
```
user@dev:~/src/securedrop$ git diff
diff --git a/securedrop/models.py b/securedrop/models.py
index ad991e6c3..1c954b9a6 100644
--- a/securedrop/models.py
+++ b/securedrop/models.py
@@ -658,6 +658,7 @@ class Journalist(db.Model):
 
         try:
             user = Journalist.query.filter_by(username=username).one()
+            return user
         except NoResultFound:
             raise InvalidUsernameException(
                 "invalid username '{}'".format(username))
```

**Review by emkll — CHANGES_REQUESTED** [review]
Thanks for the docs updates @creviera , went through the instructions and produced all the cassettes required for the functional tests :tada: 

One minor comment inline, and a rebase on latest `main` due to merge rules, and these changes should be good to merge.

File: README.md (hunk 1 above)
Comment (creviera) [reply README.md]: nice catch. i also added a gist with the patch to make it easier to download and apply.

File: README.md [hunk 2]
```diff
@@ -276,55 +276,80 @@ An AppArmor profile is available for mandatory access control. When installing s
...
-2. Start the dev server: `NUM_SOURCES=0 make dev`
-3. Set up the dev server with data required for functional tests to pass and run in any order:
-    - Create two new sources, each with one submission that contains both a file and a message. The message should be set to `this is the message`. The file should be called `hello.txt` and contain a single line of text: `hello`.
-3. Delete all the old cassettes by running `rm -r tests/functional/cassettes` or just delete the cassettes you wish to regenerate.
-4. Run `make test-functional` which will regenerate cassettes for any that are missing in the `tests/functional/cassettes` folder.
+To set up a local dev server and generate cassettes, follow these steps:
+
+1. Bypass TOTP verification so that we can use the TOTP value of `123456` hard-coded in `tests/conftest.py`. You can do this by applying the following patch to the server code:
+
+https://gist.github.com/creviera/8793d5ec4d28f034f2c1e8320a93866a
```
Comment (emkll) [inline README.md:333]: This is a good idea, definitely more readable and easier to review/use, provided the author does not delete it

**Review by emkll — APPROVED** [review]
LGTM, thanks!

**Pull request merged**